### PR TITLE
release: update manifest and helm charts for 0.0.16

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: charts/index.yaml [hunk 1]
```diff
@@ -9,7 +31,8 @@ entries:
       name: secrets-store-csi-driver
       repository: https://raw.githubusercontent.com/kubernetes-sigs/secrets-store-csi-driver/master/charts
       version: 0.0.22
-    description: A Helm chart to install the Secrets Store CSI Driver and the Azure Keyvault Provider inside a Kubernetes cluster.
+    description: A Helm chart to install the Secrets Store CSI Driver and the Azure
+      Keyvault Provider inside a Kubernetes cluster.
```
Comment (nilekhc) [inline charts/index.yaml:35]: Is this line break intentional? I could see the same on other description fields too.

File: deployment/provider-azure-installer.yaml [hunk 2]
```diff
@@ -47,6 +47,11 @@ spec:
             limits:
               cpu: 50m
               memory: 100Mi
+          securityContext:
```
Comment (nilekhc) [inline deployment/provider-azure-installer.yaml:50]: Should we be also setting `allowPrivilegeEscalation: false`?

Also `readOnlyRootFilesystem: true`. But not sure if this impedes provider's ability to write secrets on FS

File: charts/index.yaml (hunk 1 above)
Comment (aramase) [reply charts/index.yaml:35]: The line breaks are generated by `helm` as part of `helm package` command.

File: deployment/provider-azure-installer.yaml (hunk 2 above)
Comment (aramase) [reply deployment/provider-azure-installer.yaml:50]: I think we should open an issue for this and track it for the next release to ensure we get enough test results with the changes. WDYT?

Comment (nilekhc) [reply deployment/provider-azure-installer.yaml:50]: Sounds good.

Comment (aramase) [reply deployment/provider-azure-installer.yaml:50]: Created https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/537

**Review by nilekhc — APPROVED** [review]
lgtm

**Pull request merged**